lib/passkeys_signer.dart

```diff
@@ -0,0 +1,96 @@
+import 'dart:convert';
+
+import 'package:dfns_sdk_flutter/datamodel/dfns_api.dart';
+import 'package:passkeys/authenticator.dart';
+import 'package:passkeys/types.dart';
+
+class PasskeysSigner {
+  static Future<Fido2Attestation> register(
+      UserRegistrationChallenge challenge) async {
+    final registerResponse = await PasskeyAuthenticator().register(
+      RegisterRequestType(
+        challenge: challenge.challenge,
+        relyingParty: RelyingPartyType(
+          name: challenge.rp.name,
+          id: challenge.rp.id,
+        ),
+        user: UserType(
+          displayName: challenge.user.displayName,
+          name: challenge.user.name,
+          id: base64UrlEncode(utf8.encode(challenge.user.id)),
+        ),
+        authSelectionType: AuthenticatorSelectionType(
+          authenticatorAttachment:
+              challenge.authenticatorSelection.authenticatorAttachment ??
+                  'platform',
+          requireResidentKey:
+              challenge.authenticatorSelection.requireResidentKey,
+          residentKey: challenge.authenticatorSelection.residentKey,
+          userVerification: challenge.authenticatorSelection.userVerification,
+        ),
+        pubKeyCredParams: List<PubKeyCredParamType>.from(
+          challenge.pubKeyCredParams.map(
+            (e) => PubKeyCredParamType(
+              type: e.type,
+              alg: e.alg,
+            ),
+          ),
+        ),
+        timeout: null,
```

Comment:
for react native, I made this [a configurable option](https://github.com/dfns/dfns-sdk-ts/blob/m/packages/sdk-react-native/index.ts#L40).

Reply:
what I mean is make the class not static, and on construction, take in an option that configures the timeout.